planner/core/preprocess.go

```diff
@@ -1685,3 +1697,103 @@ func (p *preprocessor) hasAutoConvertWarning(colDef *ast.ColumnDef) bool {
 	}
 	return false
 }
+
+func tryLockMDLAndUpdateSchemaIfNecessary(sctx sessionctx.Context, dbName model.CIStr, tbl table.Table, is infoschema.InfoSchema) (table.Table, error) {
```

Comment:
if this can work for partition table， I think current for partition table we only lock table id, not partiont table id?

Reply:
We don't need to consider partition id.